src/calibre/gui2/main.py

```diff
@@ -400,11 +396,11 @@ def run_gui(opts, args, listener, app, gui_debug=None):
         if getattr(runner.main, 'debug_on_restart', False) or gui_debug is not None:
             run_in_debug_mode()
         else:
-            import subprocess
+            import shlex, subprocess
             if hasattr(sys, 'frameworks_dir'):
                 app = os.path.dirname(os.path.dirname(os.path.realpath(sys.frameworks_dir)))
                 prints('Restarting with:', app)
-                subprocess.Popen('sleep 3s; open ' + shellquote(app), shell=True)
+                subprocess.Popen('sleep 3s; open ' + shlex.quote(app), shell=True)
```

Comment:
That's probably some old debugging stuff left over. I'd remove the sleep.
Then this should simply be: `subprocess.Popen(app)`.
(But this patch is OK. This can be simplified later.)